packages/alchemy/src/provider/__tests__/light-account.test.ts

```diff
@@ -132,6 +132,23 @@ describe("Alchemy Provider Tests", () => {
       '"Alchemy SDK client JSON-RPC URL must match AlchemyProvider JSON-RPC URL"'
     );
   });
+
+  it("should correctly do runtime validation when chain is not supported by Alchemy", async () => {
+    expect(() => {
+      givenConnectedProvider({ owner, chain: avalanche });
+    }).toThrowErrorMatchingInlineSnapshot(`
+      "[
+        {
+          \\"code\\": \\"custom\\",
+          \\"message\\": \\"chain is not supported by Alchemy\\",
```

Comment:
Do we want to specify that this is based on viem supported chains?

Reply:
I think it's fine not to